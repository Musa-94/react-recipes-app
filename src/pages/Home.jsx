import React, { PureComponent } from 'react';
import Header from '../components/Header.jsx';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

class Home extends PureComponent {
    render() {
        return (
            <Header
                title={'amazing recipes'}
                children={
                    <Link
                        to={'/recipes'}
                        className={'text-uppercase btn btn-secondary btn-lg mt-3'}
                        children={'search recipes'}
                    />
                }
            />
        );
    }
}

Home.propTypes = {};

export default Home;
