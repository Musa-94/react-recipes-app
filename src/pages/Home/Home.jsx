import React from 'react';
import Header from '../../components/Header/Header.jsx';

const Home = () => {
    return (
        <Header
            title={'amazing recipes'}
            children={
                <a
                    href={'#/recipes'}
                    children={'search recipes'}
                    className={'text-uppercase btn btn-secondary btn-lg mt-3'}
                />
            }
        />
    );
}

export default React.memo(Home);
