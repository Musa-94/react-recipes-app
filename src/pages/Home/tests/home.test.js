import React from 'react';
import Home from '../Home';

describe('Home snapshot', () => {
    it('should correct renderer', () => {
        const component = renderer(<Home/>);

        expect(component).toMatchSnapshot();
    });
});