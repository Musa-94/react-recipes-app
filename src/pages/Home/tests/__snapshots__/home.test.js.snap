// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot should correct renderer 1`] = `
<header>
  <div
    className="container-fluid"
  >
    <div
      className="row align-items-center header-hero"
    >
      <div
        className="col text-center text-slanted"
      >
        <h1
          className="text-light text-uppercase display-3 letter-spacing text-slander"
        >
          amazing recipes
        </h1>
        <a
          className="text-uppercase btn btn-secondary btn-lg mt-3"
          href="#/recipes"
        >
          search recipes
        </a>
      </div>
    </div>
  </div>
</header>
`;
